client/flutter/pubspec.lock

```diff
@@ -7,42 +7,42 @@ packages:
       name: archive
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.0.11"
+    version: "2.0.13"
```

Comment:
Just to make sure -- these versions were all automatically incremented when you added the firebase_analytics plugin, right?

Reply:
Not sure, why would it be an issue?

Reply:
how does the `run git diff --exit-code` stage in CI work?

Reply:
> Not sure, why would it be an issue?

It's just more risk when we update many dependencies at once, especially if one of them has breaking changes. If the dependency update was required in order to use firebase_analytics, it should be fine though.

> how does the `run git diff --exit-code` stage in CI work?

The CI builds the app on iOS and Android. When build for iOS, the Podfile and Podfile.lock files are also changed if we add more dependencies to pubspec.lock. The `git diff` command in CI makes sure that we have also built the iOS version locally, so that we check in the additional changes that should be made to Podfile and Podfile.lock.

Reply:
https://dart.dev/tools/pub/versioning#semantic-versions
It shouldn't be risky as semantic versioning should prevent breaking changes from being pushed without manually changing the version constraints in `pubspec.yaml`. In fact, by not automatically accepting minor upgrades we risk missing out on security patches.

I don't build on mac so updating Podfiles isn't ideal.

Reply:
That's not necessarily true -- `boolean_selector` was upgraded from 1 to 2 -- apparently a breaking change.

Reply:
It's also a **transitive** dependency so that's not an issue.

Reply:
We don't specify it in `pubspec.yaml`. Another package uses it and want's to use the latest version. By not letting it update, multiple packages would be outdated.

Reply:
> Another package uses it and want's to use the latest version.

right -- that was my initial quesition -- "these versions were all automatically incremented when you added the firebase_analytics plugin, right?"

Reply:
Yes but some of them may have been incremented before I added the `firebase_crashlytics` dependency so I can't be certain.

Reply:
I didn't manually change `pubspec.lock`